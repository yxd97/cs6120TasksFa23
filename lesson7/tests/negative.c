#include <stdio.h>
#include <stdlib.h>
// ARGS: 3
int main(int argc, char** argv) {
    if (argc < 2) {
        return 1;
    }
    int x = atoi(argv[1]);
    int r1 = x * -3;
    printf("%d x %d = %d\n", x, -3, r1);
    int r2 = x * -245;
    printf("%d x %d = %d\n", x, -245, r2);
    int r3 = x * -63;
    printf("%d x %d = %d\n", x, -63, r3);
    int r4 = x * -1;
    printf("%d x %d = %d\n", x, -1, r4);
    int r5 = x * -43959688;
    printf("%d x %d = %d\n", x, -43959688, r5);
    return 0;
}