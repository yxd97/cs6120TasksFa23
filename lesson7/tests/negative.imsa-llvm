; ModuleID = 'tests/negative.c'
source_filename = "tests/negative.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [14 x i8] c"%d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %12 = load i32, ptr %4, align 4
  %13 = icmp slt i32 %12, 2
  br i1 %13, label %14, label %15

14:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %323

15:                                               ; preds = %2
  %16 = load ptr, ptr %5, align 8
  %17 = getelementptr inbounds ptr, ptr %16, i64 1
  %18 = load ptr, ptr %17, align 8
  %19 = call i32 @atoi(ptr noundef %18) #3
  store i32 %19, ptr %6, align 4
  %20 = load i32, ptr %6, align 4
  %21 = mul nsw i32 %20, -3
  %22 = shl i32 %20, 0
  %23 = add i32 0, %22
  %24 = shl i32 %20, 2
  %25 = add i32 %23, %24
  %26 = shl i32 %20, 3
  %27 = add i32 %25, %26
  %28 = shl i32 %20, 4
  %29 = add i32 %27, %28
  %30 = shl i32 %20, 5
  %31 = add i32 %29, %30
  %32 = shl i32 %20, 6
  %33 = add i32 %31, %32
  %34 = shl i32 %20, 7
  %35 = add i32 %33, %34
  %36 = shl i32 %20, 8
  %37 = add i32 %35, %36
  %38 = shl i32 %20, 9
  %39 = add i32 %37, %38
  %40 = shl i32 %20, 10
  %41 = add i32 %39, %40
  %42 = shl i32 %20, 11
  %43 = add i32 %41, %42
  %44 = shl i32 %20, 12
  %45 = add i32 %43, %44
  %46 = shl i32 %20, 13
  %47 = add i32 %45, %46
  %48 = shl i32 %20, 14
  %49 = add i32 %47, %48
  %50 = shl i32 %20, 15
  %51 = add i32 %49, %50
  %52 = shl i32 %20, 16
  %53 = add i32 %51, %52
  %54 = shl i32 %20, 17
  %55 = add i32 %53, %54
  %56 = shl i32 %20, 18
  %57 = add i32 %55, %56
  %58 = shl i32 %20, 19
  %59 = add i32 %57, %58
  %60 = shl i32 %20, 20
  %61 = add i32 %59, %60
  %62 = shl i32 %20, 21
  %63 = add i32 %61, %62
  %64 = shl i32 %20, 22
  %65 = add i32 %63, %64
  %66 = shl i32 %20, 23
  %67 = add i32 %65, %66
  %68 = shl i32 %20, 24
  %69 = add i32 %67, %68
  %70 = shl i32 %20, 25
  %71 = add i32 %69, %70
  %72 = shl i32 %20, 26
  %73 = add i32 %71, %72
  %74 = shl i32 %20, 27
  %75 = add i32 %73, %74
  %76 = shl i32 %20, 28
  %77 = add i32 %75, %76
  %78 = shl i32 %20, 29
  %79 = add i32 %77, %78
  %80 = shl i32 %20, 30
  %81 = add i32 %79, %80
  %82 = shl i32 %20, 31
  %83 = add i32 %81, %82
  store i32 %83, ptr %7, align 4
  %84 = load i32, ptr %6, align 4
  %85 = load i32, ptr %7, align 4
  %86 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %84, i32 noundef -3, i32 noundef %85)
  %87 = load i32, ptr %6, align 4
  %88 = mul nsw i32 %87, -245
  %89 = shl i32 %87, 0
  %90 = add i32 0, %89
  %91 = shl i32 %87, 1
  %92 = add i32 %90, %91
  %93 = shl i32 %87, 3
  %94 = add i32 %92, %93
  %95 = shl i32 %87, 8
  %96 = add i32 %94, %95
  %97 = shl i32 %87, 9
  %98 = add i32 %96, %97
  %99 = shl i32 %87, 10
  %100 = add i32 %98, %99
  %101 = shl i32 %87, 11
  %102 = add i32 %100, %101
  %103 = shl i32 %87, 12
  %104 = add i32 %102, %103
  %105 = shl i32 %87, 13
  %106 = add i32 %104, %105
  %107 = shl i32 %87, 14
  %108 = add i32 %106, %107
  %109 = shl i32 %87, 15
  %110 = add i32 %108, %109
  %111 = shl i32 %87, 16
  %112 = add i32 %110, %111
  %113 = shl i32 %87, 17
  %114 = add i32 %112, %113
  %115 = shl i32 %87, 18
  %116 = add i32 %114, %115
  %117 = shl i32 %87, 19
  %118 = add i32 %116, %117
  %119 = shl i32 %87, 20
  %120 = add i32 %118, %119
  %121 = shl i32 %87, 21
  %122 = add i32 %120, %121
  %123 = shl i32 %87, 22
  %124 = add i32 %122, %123
  %125 = shl i32 %87, 23
  %126 = add i32 %124, %125
  %127 = shl i32 %87, 24
  %128 = add i32 %126, %127
  %129 = shl i32 %87, 25
  %130 = add i32 %128, %129
  %131 = shl i32 %87, 26
  %132 = add i32 %130, %131
  %133 = shl i32 %87, 27
  %134 = add i32 %132, %133
  %135 = shl i32 %87, 28
  %136 = add i32 %134, %135
  %137 = shl i32 %87, 29
  %138 = add i32 %136, %137
  %139 = shl i32 %87, 30
  %140 = add i32 %138, %139
  %141 = shl i32 %87, 31
  %142 = add i32 %140, %141
  store i32 %142, ptr %8, align 4
  %143 = load i32, ptr %6, align 4
  %144 = load i32, ptr %8, align 4
  %145 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %143, i32 noundef -245, i32 noundef %144)
  %146 = load i32, ptr %6, align 4
  %147 = mul nsw i32 %146, -63
  %148 = shl i32 %146, 0
  %149 = add i32 0, %148
  %150 = shl i32 %146, 6
  %151 = add i32 %149, %150
  %152 = shl i32 %146, 7
  %153 = add i32 %151, %152
  %154 = shl i32 %146, 8
  %155 = add i32 %153, %154
  %156 = shl i32 %146, 9
  %157 = add i32 %155, %156
  %158 = shl i32 %146, 10
  %159 = add i32 %157, %158
  %160 = shl i32 %146, 11
  %161 = add i32 %159, %160
  %162 = shl i32 %146, 12
  %163 = add i32 %161, %162
  %164 = shl i32 %146, 13
  %165 = add i32 %163, %164
  %166 = shl i32 %146, 14
  %167 = add i32 %165, %166
  %168 = shl i32 %146, 15
  %169 = add i32 %167, %168
  %170 = shl i32 %146, 16
  %171 = add i32 %169, %170
  %172 = shl i32 %146, 17
  %173 = add i32 %171, %172
  %174 = shl i32 %146, 18
  %175 = add i32 %173, %174
  %176 = shl i32 %146, 19
  %177 = add i32 %175, %176
  %178 = shl i32 %146, 20
  %179 = add i32 %177, %178
  %180 = shl i32 %146, 21
  %181 = add i32 %179, %180
  %182 = shl i32 %146, 22
  %183 = add i32 %181, %182
  %184 = shl i32 %146, 23
  %185 = add i32 %183, %184
  %186 = shl i32 %146, 24
  %187 = add i32 %185, %186
  %188 = shl i32 %146, 25
  %189 = add i32 %187, %188
  %190 = shl i32 %146, 26
  %191 = add i32 %189, %190
  %192 = shl i32 %146, 27
  %193 = add i32 %191, %192
  %194 = shl i32 %146, 28
  %195 = add i32 %193, %194
  %196 = shl i32 %146, 29
  %197 = add i32 %195, %196
  %198 = shl i32 %146, 30
  %199 = add i32 %197, %198
  %200 = shl i32 %146, 31
  %201 = add i32 %199, %200
  store i32 %201, ptr %9, align 4
  %202 = load i32, ptr %6, align 4
  %203 = load i32, ptr %9, align 4
  %204 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %202, i32 noundef -63, i32 noundef %203)
  %205 = load i32, ptr %6, align 4
  %206 = mul nsw i32 %205, -1
  %207 = shl i32 %205, 0
  %208 = add i32 0, %207
  %209 = shl i32 %205, 1
  %210 = add i32 %208, %209
  %211 = shl i32 %205, 2
  %212 = add i32 %210, %211
  %213 = shl i32 %205, 3
  %214 = add i32 %212, %213
  %215 = shl i32 %205, 4
  %216 = add i32 %214, %215
  %217 = shl i32 %205, 5
  %218 = add i32 %216, %217
  %219 = shl i32 %205, 6
  %220 = add i32 %218, %219
  %221 = shl i32 %205, 7
  %222 = add i32 %220, %221
  %223 = shl i32 %205, 8
  %224 = add i32 %222, %223
  %225 = shl i32 %205, 9
  %226 = add i32 %224, %225
  %227 = shl i32 %205, 10
  %228 = add i32 %226, %227
  %229 = shl i32 %205, 11
  %230 = add i32 %228, %229
  %231 = shl i32 %205, 12
  %232 = add i32 %230, %231
  %233 = shl i32 %205, 13
  %234 = add i32 %232, %233
  %235 = shl i32 %205, 14
  %236 = add i32 %234, %235
  %237 = shl i32 %205, 15
  %238 = add i32 %236, %237
  %239 = shl i32 %205, 16
  %240 = add i32 %238, %239
  %241 = shl i32 %205, 17
  %242 = add i32 %240, %241
  %243 = shl i32 %205, 18
  %244 = add i32 %242, %243
  %245 = shl i32 %205, 19
  %246 = add i32 %244, %245
  %247 = shl i32 %205, 20
  %248 = add i32 %246, %247
  %249 = shl i32 %205, 21
  %250 = add i32 %248, %249
  %251 = shl i32 %205, 22
  %252 = add i32 %250, %251
  %253 = shl i32 %205, 23
  %254 = add i32 %252, %253
  %255 = shl i32 %205, 24
  %256 = add i32 %254, %255
  %257 = shl i32 %205, 25
  %258 = add i32 %256, %257
  %259 = shl i32 %205, 26
  %260 = add i32 %258, %259
  %261 = shl i32 %205, 27
  %262 = add i32 %260, %261
  %263 = shl i32 %205, 28
  %264 = add i32 %262, %263
  %265 = shl i32 %205, 29
  %266 = add i32 %264, %265
  %267 = shl i32 %205, 30
  %268 = add i32 %266, %267
  %269 = shl i32 %205, 31
  %270 = add i32 %268, %269
  store i32 %270, ptr %10, align 4
  %271 = load i32, ptr %6, align 4
  %272 = load i32, ptr %10, align 4
  %273 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %271, i32 noundef -1, i32 noundef %272)
  %274 = load i32, ptr %6, align 4
  %275 = mul nsw i32 %274, -43958
  %276 = shl i32 %274, 1
  %277 = add i32 0, %276
  %278 = shl i32 %274, 3
  %279 = add i32 %277, %278
  %280 = shl i32 %274, 6
  %281 = add i32 %279, %280
  %282 = shl i32 %274, 10
  %283 = add i32 %281, %282
  %284 = shl i32 %274, 12
  %285 = add i32 %283, %284
  %286 = shl i32 %274, 14
  %287 = add i32 %285, %286
  %288 = shl i32 %274, 16
  %289 = add i32 %287, %288
  %290 = shl i32 %274, 17
  %291 = add i32 %289, %290
  %292 = shl i32 %274, 18
  %293 = add i32 %291, %292
  %294 = shl i32 %274, 19
  %295 = add i32 %293, %294
  %296 = shl i32 %274, 20
  %297 = add i32 %295, %296
  %298 = shl i32 %274, 21
  %299 = add i32 %297, %298
  %300 = shl i32 %274, 22
  %301 = add i32 %299, %300
  %302 = shl i32 %274, 23
  %303 = add i32 %301, %302
  %304 = shl i32 %274, 24
  %305 = add i32 %303, %304
  %306 = shl i32 %274, 25
  %307 = add i32 %305, %306
  %308 = shl i32 %274, 26
  %309 = add i32 %307, %308
  %310 = shl i32 %274, 27
  %311 = add i32 %309, %310
  %312 = shl i32 %274, 28
  %313 = add i32 %311, %312
  %314 = shl i32 %274, 29
  %315 = add i32 %313, %314
  %316 = shl i32 %274, 30
  %317 = add i32 %315, %316
  %318 = shl i32 %274, 31
  %319 = add i32 %317, %318
  store i32 %319, ptr %11, align 4
  %320 = load i32, ptr %6, align 4
  %321 = load i32, ptr %11, align 4
  %322 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %320, i32 noundef -43958, i32 noundef %321)
  store i32 0, ptr %3, align 4
  br label %323

323:                                              ; preds = %15, %14
  %324 = load i32, ptr %3, align 4
  ret i32 %324
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
