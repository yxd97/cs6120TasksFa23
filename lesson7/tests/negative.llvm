; ModuleID = 'tests/negative.c'
source_filename = "tests/negative.c"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-pc-linux-gnu"

@.str = private unnamed_addr constant [14 x i8] c"%d x %d = %d\0A\00", align 1

; Function Attrs: noinline nounwind optnone uwtable
define dso_local i32 @main(i32 noundef %0, ptr noundef %1) #0 {
  %3 = alloca i32, align 4
  %4 = alloca i32, align 4
  %5 = alloca ptr, align 8
  %6 = alloca i32, align 4
  %7 = alloca i32, align 4
  %8 = alloca i32, align 4
  %9 = alloca i32, align 4
  %10 = alloca i32, align 4
  %11 = alloca i32, align 4
  store i32 0, ptr %3, align 4
  store i32 %0, ptr %4, align 4
  store ptr %1, ptr %5, align 8
  %12 = load i32, ptr %4, align 4
  %13 = icmp slt i32 %12, 2
  br i1 %13, label %14, label %15

14:                                               ; preds = %2
  store i32 1, ptr %3, align 4
  br label %45

15:                                               ; preds = %2
  %16 = load ptr, ptr %5, align 8
  %17 = getelementptr inbounds ptr, ptr %16, i64 1
  %18 = load ptr, ptr %17, align 8
  %19 = call i32 @atoi(ptr noundef %18) #3
  store i32 %19, ptr %6, align 4
  %20 = load i32, ptr %6, align 4
  %21 = mul nsw i32 %20, -3
  store i32 %21, ptr %7, align 4
  %22 = load i32, ptr %6, align 4
  %23 = load i32, ptr %7, align 4
  %24 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %22, i32 noundef -3, i32 noundef %23)
  %25 = load i32, ptr %6, align 4
  %26 = mul nsw i32 %25, -245
  store i32 %26, ptr %8, align 4
  %27 = load i32, ptr %6, align 4
  %28 = load i32, ptr %8, align 4
  %29 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %27, i32 noundef -245, i32 noundef %28)
  %30 = load i32, ptr %6, align 4
  %31 = mul nsw i32 %30, -63
  store i32 %31, ptr %9, align 4
  %32 = load i32, ptr %6, align 4
  %33 = load i32, ptr %9, align 4
  %34 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %32, i32 noundef -63, i32 noundef %33)
  %35 = load i32, ptr %6, align 4
  %36 = mul nsw i32 %35, -1
  store i32 %36, ptr %10, align 4
  %37 = load i32, ptr %6, align 4
  %38 = load i32, ptr %10, align 4
  %39 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %37, i32 noundef -1, i32 noundef %38)
  %40 = load i32, ptr %6, align 4
  %41 = mul nsw i32 %40, -43958
  store i32 %41, ptr %11, align 4
  %42 = load i32, ptr %6, align 4
  %43 = load i32, ptr %11, align 4
  %44 = call i32 (ptr, ...) @printf(ptr noundef @.str, i32 noundef %42, i32 noundef -43958, i32 noundef %43)
  store i32 0, ptr %3, align 4
  br label %45

45:                                               ; preds = %15, %14
  %46 = load i32, ptr %3, align 4
  ret i32 %46
}

; Function Attrs: nounwind willreturn memory(read)
declare i32 @atoi(ptr noundef) #1

declare i32 @printf(ptr noundef, ...) #2

attributes #0 = { noinline nounwind optnone uwtable "frame-pointer"="all" "min-legal-vector-width"="0" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #1 = { nounwind willreturn memory(read) "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #2 = { "frame-pointer"="all" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+cmov,+cx8,+fxsr,+mmx,+sse,+sse2,+x87" "tune-cpu"="generic" }
attributes #3 = { nounwind willreturn memory(read) }

!llvm.module.flags = !{!0, !1, !2, !3, !4}
!llvm.ident = !{!5}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{i32 8, !"PIC Level", i32 2}
!2 = !{i32 7, !"PIE Level", i32 2}
!3 = !{i32 7, !"uwtable", i32 2}
!4 = !{i32 7, !"frame-pointer", i32 2}
!5 = !{!"Ubuntu clang version 17.0.2 (++20230929073225+098e653a5bed-1~exp1~20230929073241.48)"}
