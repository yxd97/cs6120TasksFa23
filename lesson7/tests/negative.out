3 x -3 = -9
3 x -245 = -735
3 x -63 = -189
3 x -1 = -3
3 x -43958 = -131874
